### PR TITLE
fix: value-head require grad; override maniskill put-on-carrot-env; m…

### PR DESCRIPTION
**Pull request by zanghz21**
…odify grpo func loss weight

### Description

- Fix the bug about value-head gradient. Previously, the value-head did not have gradients under lora setting, and did not use separated learning rate when lora is False. 
- Change some loss ratio computation in grpo with mask settings.
- Override the ManiSkill PutCarrotOnPlate task to exclude the _settle function.

### Motivation and Context

- Fix ppo bug. Make ppo training stable.
- Refine grpo loss weight.
- Simplify the docker setup.

### How has this been tested?

- Run embodiment test scripts. 

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: rlinf/algorithms/embodiment/grpo_functions.py [hunk 1]
```diff
@@ -265,7 +263,6 @@ def actor_loss_fn(
             - actor/pg_clipfrac: Fraction of clipped policy gradient loss
```
Comment (QuanluZhang) [inline rlinf/algorithms/embodiment/grpo_functions.py:265]: `loss_mask` and `loss_mask_ratio` are not included in docstring

File: rlinf/models/__init__.py [hunk 2]
```diff
@@ -200,6 +200,10 @@ def get_model(model_path, cfg: DictConfig, override_config_kwargs=None):
         else:
             model = PeftModel.from_pretrained(model, cfg.lora_path, is_trainable=True)
 
+    for name, param in model.named_parameters():
+        if "value_head" in name:
+            param.requires_grad = True
```
Comment (QuanluZhang) [inline rlinf/models/__init__.py]: this part is a little hacky, which model requires this logic? If you place this logic here, it means for all the models with the name "xxxvalue_headxxx" params must be `requires_grad=True`. Is this our expected behavior?

File: rlinf/workers/actor/fsdp_actor_worker.py [hunk 3]
```diff
@@ -304,6 +304,10 @@ def run_training(self):
                 prev_values = data["prev_values"]
                 loss_mask = data.get("loss_mask", None)
                 loss_mask_sum = data.get("loss_mask_sum", None)
+                if hasattr(self.cfg.env.train, "max_episode_steps"):
+                    max_episode_steps = self.cfg.env.train.max_episode_steps
+                else:
+                    max_episode_steps = self.cfg.env.train.init_params.max_episode_steps
```
Comment (QuanluZhang) [inline rlinf/workers/actor/fsdp_actor_worker.py]: what is the difference between `train.max_episode_steps` and `train.init_prams.max_episode_steps`? different configurations in yaml?

File: rlinf/models/__init__.py (hunk 2 above)
Comment (zanghz21) [reply rlinf/models/__init__.py]: Currently, we assume that the value_head serves as the critic in the ppo setting.

File: rlinf/workers/actor/fsdp_actor_worker.py (hunk 3 above)
Comment (zanghz21) [reply rlinf/workers/actor/fsdp_actor_worker.py]: Currently in maniskill setting, this param is under init_params, in libero setting, this param is directly under env.train. The contents under init_params are determined by the simulator. Maybe we can add a new attr under env.train in the maniskill setting, but this requires modification in yaml.

Comment (zoeyuchao) [reply rlinf/workers/actor/fsdp_actor_worker.py]: I support adding a new attribute in the maniskill setting. It is okay to modify maniskill yaml.

File: rlinf/models/__init__.py (hunk 2 above)
Comment (zoeyuchao) [reply rlinf/models/__init__.py]: It is okay here, the name of value_head is not used that much in other settings, but maybe we can leave a comment here, say, Attn: make sure the name of value head only means the critic of PPO.

File: examples/embodiment/config/maniskill_ppo_openvla_quickstart.yaml [hunk 4]
```diff
@@ -36,6 +36,7 @@ runner:
 algorithm:
   auto_reset: True
   ignore_terminations: True
+  use_fixed_reset_state_ids: False
```
Comment (zoeyuchao) [inline examples/embodiment/config/maniskill_ppo_openvla_quickstart.yaml:39]: maybe we can add some description about this term, what does it control?

**Review by zoeyuchao — APPROVED** [review]
I have reviewed all changes and leave comments, only minor comments are needed, soi approved this pr. 
Good Job, Hongzhi!

**Review by XuS1994 — APPROVED** [review]

File: examples/embodiment/config/maniskill_ppo_openvla_quickstart.yaml (hunk 4 above)
Comment (zanghz21) [reply examples/embodiment/config/maniskill_ppo_openvla_quickstart.yaml:39]: Add it in the doc

File: rlinf/workers/actor/fsdp_actor_worker.py (hunk 3 above)
Comment (zanghz21) [reply rlinf/workers/actor/fsdp_actor_worker.py]: done

File: rlinf/models/__init__.py (hunk 2 above)
Comment (zanghz21) [reply rlinf/models/__init__.py]: add 1 comment on value-head when initialize the optimizer params group

File: rlinf/algorithms/embodiment/grpo_functions.py (hunk 1 above)
Comment (zanghz21) [reply rlinf/algorithms/embodiment/grpo_functions.py:265]: done
